drivers/lora/Kconfig.sx12xx

```
@@ -4,49 +4,30 @@
 # SPDX-License-Identifier: Apache-2.0
 #
 
-menuconfig LORA_SX12XX
-	bool "Semtech SX-series driver"
-	select HAS_SEMTECH_RADIO_DRIVERS
-	depends on SPI
-	help
-	  Enable LoRa driver for Semtech SX12xx.
-
-if LORA_SX12XX
-
-choice
-	prompt "LoRa Radio chipset"
-	default LORA_SX127X if DT_HAS_SEMTECH_SX1272_ENABLED
-	default LORA_SX127X if DT_HAS_SEMTECH_SX1276_ENABLED
-	default LORA_SX126X if DT_HAS_SEMTECH_SX1261_ENABLED
-	default LORA_SX126X if DT_HAS_SEMTECH_SX1262_ENABLED
-	default LORA_STM32WL_SUBGHZ_RADIO if DT_HAS_ST_STM32WL_SUBGHZ_RADIO_ENABLED
-	help
-	  Select the LoRa modem used on your board. The default value
-	  is discovered from the device tree and should be correct for
-	  most users.
-
 config LORA_SX127X
 	bool "Semtech SX127x driver"
+	default y
 	depends on DT_HAS_SEMTECH_SX1272_ENABLED || DT_HAS_SEMTECH_SX1276_ENABLED
 	select HAS_SEMTECH_SX1272 if DT_HAS_SEMTECH_SX1272_ENABLED
 	select HAS_SEMTECH_SX1276 if DT_HAS_SEMTECH_SX1276_ENABLED
+	depends on SPI
```

Should these be:
```suggestion
	select SPI
	select GPIO
```

Then the `CONFIG_SPI=y` and `CONFIG_GPIO=y` lines can also be potentially removed from `prj.conf` files.

---

That is potentially another change to make, but I would be looking to do a complete pass of all usages of `depends on SPI` to clean up inconsistencies. It's not the primary purpose of this PR.